test/module/irohad/ametsuchi/settings_test.cpp

```diff
@@ -0,0 +1,114 @@
+
+        executor = std::make_unique<PostgresCommandExecutor>(
+            std::make_unique<soci::session>(*soci::factory_postgresql(),
+                                            pgopt_),
+            perm_converter);
+
+        setting_query = std::make_unique<PostgresSettingQuery>(
+            std::make_unique<soci::session>(*soci::factory_postgresql(),
+                                            pgopt_),
+            getTestLogger("SettingQuery"));
+      }
+
+      /**
+       * Execute a given command and optionally check its result
+       * @tparam CommandType - type of the command
+       * @param command - the command to CHECK_SUCCESSFUL_RESULT(execute
+       * @param do_validation - of the command should be validated
+       * @param creator - creator of the command
+       * @return result of command execution
+       */
+      template <typename CommandType>
+      CommandResult execute(CommandType &&command,
+                            bool do_validation = false,
+                            const shared_model::interface::types::AccountIdType
+                                &creator = "id@domain") {
+        // TODO igor-egorov 15.04.2019 IR-446 Refactor postgres_executor_test
+        shared_model::interface::Command::CommandVariantType variant{
+            std::forward<CommandType>(command)};
+        shared_model::interface::MockCommand cmd;
+        EXPECT_CALL(cmd, get()).WillRepeatedly(::testing::ReturnRef(variant));
+        return executor->execute(cmd, creator, not do_validation);
+      }
+
+      void TearDown() override {
+        AmetsuchiTest::TearDown();
+      }
+
+      std::unique_ptr<CommandExecutor> executor;
+      std::shared_ptr<shared_model::interface::PermissionToString>
+          perm_converter =
+              std::make_shared<shared_model::proto::ProtoPermissionToString>();
+
+      std::unique_ptr<SettingQuery> setting_query;
+      std::unique_ptr<shared_model::interface::MockCommandFactory>
+          mock_command_factory =
+              std::make_unique<shared_model::interface::MockCommandFactory>();
+    };
+
+    /**
+     * @given settings
+     * @when trying to get setting with the key that doesn't exist
+     * @then settings return default value
+     */
+    TEST_F(SettingsTest, NoSuchSetting) {
+      auto settings = setting_query->get();
+      ASSERT_EQ(settings->max_description_size, 64);
+    }
+
+    /**
+     * @given settings
+     * @when trying to get setting with the key that has invalid value
+     * @then settings return default value
+     */
+    TEST_F(SettingsTest, InvalidSettingValue) {
+      std::string value = "two";
+      execute(*mock_command_factory->constructSetSettingValue(
+                  shared_model::validation::Settings::kMaxDescriptionSizeKey,
+                  value),
+              true);
+
+      std::shared_ptr<const shared_model::validation::Settings> settings;
+      EXPECT_ANY_THROW(settings = setting_query->get());
```

Comment:
`setting_query` should not throw exceptions.

Reply:
Fixed